client/src/main/java/io/pravega/client/stream/ReaderGroup.java

```diff
@@ -131,7 +130,7 @@
      * A stream cut corresponds to a position in the stream, and it can be used by
      * the application as reference to such a position.
      *
-     * @return Map of streams that this group is reading from to the corresponding cuts.
+     * @return {@link StreamCut StreamCuts} of streams that this reader group is reading from.
      */
-    Map<Stream, StreamCut> getStreamCuts();
```

Comment:
The reason why this is like this is because in `CheckpointImpl` we have a map and not a set. Having a `map` is consistent with the implementation of checkpointing. We don't necessarily need to use the same data structure, but I'd like to understand why we have a `map` there while we can have a `set`here. Please clarify.

Reply:
As per my current understanding there is no specific need to have a `Map<Stream, StreamCut>` inside the `CheckPointImpl`.  `CheckPointImpl` when passed to `resetReadersToCheckpoint()` internally uses only the the values of the map i.e. essentially a Collection of stream cuts .
Also, `StreamCut` internally represents the `Stream` it is associated with.

Hence, we can essentially have a `Set<StreamCut>` inside `CheckPointImpl` and have the same representation for `getStreamCut()` and `resetToStreamCut()` apis.

Reply:
No, the application may want to know which is which. IE: suppose they are going to make a call on the batch api passing a stream cut. Without the map it's just awkward.

Reply:
Will retain the use of Map<Stream, StreamCut> to simplify the usage the pattern.
updated the StreamSeek API to use a Map<Stream, StreamCut> in PR #2304.